src/Ilios/CliBundle/Command/SendTeachingRemindersCommand.php

```diff
@@ -0,0 +1,251 @@
+
+        if ($offerings->isEmpty()) {
+            $output->writeln('<info>No offerings with pending teaching reminders found.</info>');
+            return;
+        }
+
+        // mail out a reminder per instructor per offering.
+        $templateCache = [];
+        $iterator = $offerings->getIterator();
+        $i = 0;
+
+        /** @var OfferingInterface $offering */
+        foreach ($iterator as $offering) {
+            $school = $offering->getSession()->getCourse()->getSchool();
+            if (! array_key_exists($school->getId(), $templateCache)) {
+                $template = $this->getTemplatePath($school);
+                $templateCache[$school->getId()] = $template;
+            }
+            $template = $templateCache[$school->getId()];
+
+            $instructors = $this->getAllInstructorsForOffering($offering);
+
+            /** @var UserInterface $instructor */
+            foreach ($instructors as $instructor) {
+                $i++;
+                $messageBody = $this->templatingEngine->render($template, [
+                    'base_url' => $baseUrl,
+                    'instructor' => $instructor,
+                    'offering' => $offering,
+                ]);
+                $message = \Swift_Message::newInstance()
+                    ->setSubject($subject)
+                    ->setFrom($sender)
+                    ->setTo($instructor->getEmail())
+                    ->setCharset('UTF-8')
+                    ->setContentType('text/plain')
+                    ->setBody($messageBody)
+                    ->setMaxLineLength(998);
+                if ($isDryRun) {
+                    $mailHeaders = $message->getHeaders()->toString();
+                    $mailBody = $message->getBody();
+                    $output->writeln("__MESSAGE_START__");
+                    $output->writeln("__HEADERS_START__\n{$mailHeaders}\n__HEADERS_END__");
+                    $output->writeln("__BODY_START__\n{$mailBody}\n__BODY_END__");
+                    $output->writeln("__MESSAGE_END__");
+                } else {
+                    $this->mailer->send($message);
+                }
+            }
+        }
+
+        $output->writeln("<info>Sent {$i} teaching reminders.</info>");
+    }
+
+    /**
+     * Locates the applicable message template for a given school and returns its path.
+     * @param SchoolInterface $school
+     * @return string The template path.
+     */
+    protected function getTemplatePath(SchoolInterface $school)
+    {
+        $paths = [
+            '@custom_email_templates/' . basename($school->getTemplatePrefix() . '_' . self::DEFAULT_TEMPLATE_NAME),
+            '@custom_email_templates/' . self::DEFAULT_TEMPLATE_NAME,
+        ];
+        foreach ($paths as $path) {
+            if ($this->templatingEngine->exists($path)) {
+                return $path;
+            }
+        }
+        return 'IliosCoreBundle:Email:' .self::DEFAULT_TEMPLATE_NAME;
+    }
```

Comment:
The is clean!  I like the fallback and the school specific templates, very nice.